client/me/purchases/purchases-list-in-dataviews/index.tsx

```diff
@@ -0,0 +1,204 @@
+	NextAppointment,
+} from 'calypso/state/selectors/get-concierge-next-appointment';
+import getConciergeUserBlocked from 'calypso/state/selectors/get-concierge-user-blocked';
+import getSites from 'calypso/state/selectors/get-sites';
+import { getSiteId } from 'calypso/state/sites/selectors';
+import { AppState } from 'calypso/types';
+import MembershipSite from '../membership-site';
+import PurchasesSite from '../purchases-site';
+import { purchasesDataFields } from './purchases-data-field';
+import { purchasesDataView } from './purchases-data-view'; // This is the temporary data
+import PurchasesListHeader from './purchases-list-header';
+import { testoctorPurchases } from './tests/testoctor-flat-data'; // This is the temporary data
+
+const useDataViewPurchasesList = config.isEnabled( 'purchases/purchase-list-dataview' );
+
+export interface PurchasesListProps {
+	noticeType?: string | undefined;
+}
+
+export interface PurchasesListConnectedProps {
+	hasLoadedUserPurchasesFromServer: boolean;
+	isFetchingUserPurchases: boolean;
+	purchases: Purchase[] | null;
+	subscriptions: MembershipSubscription[];
+	sites: SiteDetails[];
+	nextAppointment: NextAppointment | null;
+	isUserBlocked: boolean;
+	availableSessions: number[];
+	siteId: number | null;
+}
+
+class PurchasesListDataView extends Component<
+	PurchasesListProps & PurchasesListConnectedProps & WithStoredPaymentMethodsProps & LocalizeProps
+> {
+	isDataLoading() {
+		if ( this.props.isFetchingUserPurchases && ! this.props.hasLoadedUserPurchasesFromServer ) {
+			return true;
+		}
+
+		return ! this.props.sites.length && ! this.props.subscriptions.length;
+	}
+
+	renderConciergeBanner() {
+		const { nextAppointment, availableSessions, isUserBlocked } = this.props;
+		return (
+			<PurchaseListConciergeBanner
+				nextAppointment={ nextAppointment ?? undefined }
+				availableSessions={ availableSessions }
+				isUserBlocked={ isUserBlocked }
+			/>
+		);
+	}
+
+	renderMembershipSubscriptions() {
+		const { subscriptions } = this.props;
+
+		if ( ! subscriptions.length || this.isDataLoading() ) {
+			return null;
+		}
+
+		return getSubscriptionsBySite( subscriptions ).map( ( site ) => (
+			<MembershipSite site={ site } key={ site.id } />
+		) );
+	}
+
+	render() {
+		const { purchases, sites, translate, subscriptions } = this.props;
+		const commonEventProps = { context: 'me' };
+		let content;
+
+		if ( this.isDataLoading() ) {
+			content = <PurchasesSite isPlaceholder />;
+		}
+
+		const onChangeView = () => {
+			alert( 'You clicked something!!' );
+		};
+
+		if ( purchases && purchases.length ) {
+			if ( useDataViewPurchasesList ) {
+				content = (
+					<DataViews
+						data={ purchases }
+						fields={ purchasesDataFields }
+						view={ purchasesDataView }
+						onChangeView={ onChangeView }
+						defaultLayouts={ { table: {} } }
+						actions={ undefined }
+						paginationInfo={ { totalItems: 100, totalPages: 10 } }
+					/>
+				);
+			} else {
+				content = (
```

Comment:
I haven't gotten rid of the original `Active upgrades` code here. Even though it duplicates the code in `purchases-list/index.tsx`, it makes for a quick and dirty reference when I am trying to figure out how the current list in constructed.